src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.Shared.xml

```diff
@@ -19,6 +19,9 @@
     <type fullname="System.Threading.Timer">
       <property name="AllTimers" />
     </type>
+    <type fullname="System.Threading.Thread">
```

Comment:
Since this is a workaround for a CoreCLR specific issue, it would fit better into `src\coreclr\System.Private.CoreLib\src\ILLink\ILLink.Descriptors.xml`